src/platform_linux/platform.c

```diff
@@ -124,8 +153,17 @@ platform_buffer_init(buffer_handle *bh, size_t length)
          platform_error_log("mlock (%lu bytes) failed with error: %s\n",
                             length,
                             strerror(errno));
-         munmap(bh->addr, length);
+         // Save off rc from mlock()-failure, so we return that as status
          rc = CONST_STATUS(errno);
+
+         int rv = munmap(bh->addr, length);
+         if (rv != 0) {
+            // We are losing rc from this failure; can't return 2 statuses
+            platform_error_log("munmap %p (%lu bytes) failed with error: %s\n",
```

Comment:
Potential missing `rc` issue also in `/main`. Hard to repro ... figured this out from code-walk through.